### PR TITLE
Add Cropping layers support

### PR DESCRIPTION
**Pull request by HamzaEzzRa**
# Description

This PR adds support for Keras' [Cropping1D](https://keras.io/api/layers/reshaping_layers/cropping1d) and [Cropping2D](https://keras.io/api/layers/reshaping_layers/cropping2d) layers with Vivado and Vitis backend in both `io_parallel` and `io_stream` modes.

The implementation is straightforward and follows a similar style to zero-padding implementations. I added tests for Vivado and Vitis backends.

I have yet to run this on a board, but implementation was successful targeting a ZCU102, using Vivado 2019.1. Although this type of layer has a fairly niche usage, I still found a use case for it, and it could benefit the community.

Linked Issue: #1308

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Tests

I added unit tests in `test/pytest/test_cropping.py`

**Test Configuration**:
- `io_type`: `["io_stream", "io_parallel"]`
- `backend`: `["Vivado", "Vitis"]`
- `model_type`: `["1d", "2d"]`

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/fastmachinelearning/hls4ml/blob/main/CONTRIBUTING.md).
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have installed and run `pre-commit` on the files I edited or added.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by HamzaEzzRa** [comment]
pre-commit.ci autofix

**Review by JanFSchulte — APPROVED** [review]
Test error are unrelated to this PR. Looks good to me.

**Comment by JanFSchulte** [comment]
@calad0i Do you think this needs to be included explicitly also in the keras v3 parser, or should we let the fallback take care of it?
